test/specs/controller.line.tests.js

```diff
@@ -573,6 +573,96 @@ describe('Chart.controllers.line', function() {
 		expect(meta.dataset._model.borderWidth).toBe(0.55);
 	});
 
+	describe('dataset defaults', function() {
+		beforeEach(function() {
+			this._defaults = Chart.helpers.clone(Chart.defaults.global.datasets.line);
+		});
+
+		afterEach(function() {
+			Chart.defaults.global.datasets.line = this._defaults;
+			delete this._defaults;
+		});
+
+		it('should utilize the dataset default options', function() {
+			Chart.defaults.global.datasets.line = Chart.defaults.global.datasets.line || {};
+			var defaults = Chart.defaults.global.datasets.line;
+			defaults.spanGaps = true;
+			defaults.tension = 0.231;
+			defaults.backgroundColor = '#add';
+			defaults.borderWidth = '#daa';
+			defaults.borderColor = '#dad';
+			defaults.borderCapStyle = 'round';
+			defaults.borderDash = [0];
+			defaults.borderDashOffset = 0.871;
+			defaults.borderJoinStyle = 'miter';
+			defaults.fill = 'start';
+			defaults.cubicInterpolationMode = 'monotone';
+
+			var chart = window.acquireChart({
+				type: 'line',
+				data: {
+					datasets: [{
+						data: [0, 0],
+						label: 'dataset1'
+					}],
+					labels: ['label1', 'label2']
+				}
+			});
+
+			var model = chart.getDatasetMeta(0).dataset._model;
+
+			expect(model.spanGaps).toBe(true);
+			expect(model.tension).toBe(0.231);
+			expect(model.backgroundColor).toBe('#add');
+			expect(model.borderWidth).toBe('#daa');
+			expect(model.borderColor).toBe('#dad');
+			expect(model.borderCapStyle).toBe('round');
+			expect(model.borderDash).toEqual([0]);
+			expect(model.borderDashOffset).toBe(0.871);
+			expect(model.borderJoinStyle).toBe('miter');
+			expect(model.fill).toBe('start');
+			expect(model.cubicInterpolationMode).toBe('monotone');
+		});
+	});
+
+	it('should obey the dataset options', function() {
```

Comment:
This is meant to test that the dataset options apply or is it meant to test that it overrides the dataset defaults? If it is the latter, does this test needs some additional setup to provide a full set of defaults to override?

Reply:
It was meant to test the former. I can change it to the latter or add a second test to test the latter if you'd like. The latter is currently tested with one option in the scatter test. I figured that was probably sufficient since all options should behave the same.